resources/lib/youtube_plugin/youtube/helper/video_info.py

```diff
@@ -1713,3 +2178,8 @@ def filter_qualities(stream_data, container, sorted_qualities, fps_limit):
             ), stream_info
         except:
             return None, None
+        return 'http://{0}:{1}/{2}.mpd'.format(
+            _settings.httpd_listen(for_request=True),
+            _settings.httpd_port(),
+            self.video_id
+        ), main_stream
```

Comment:
This last return statement is never reached so the correct url of the generated manifest is never used as a stream option. This manifest contains all the adaptive stream formats, which is one of the reasons why only the standard format (max 720p) streams are being played.

The "success" return in the `try/except` block  should be updated as `ipaddress` and `stream_info` are not defined anywhere. This may be from an older version of the code.

Something like the following, adapted from current master, should work:
```
        filepath = '{0}{1}.mpd'.format(basepath, self.video_id)
        success = None

        try:
            mpd_file = xbmcvfs.File(filepath, 'w')
        except (IOError, OSError):
            self._context.log_error('Unable to open MPD file path for writing')
        else:
            try:
                success = mpd_file.write(str(out.encode('utf-8')))
            finally:
                mpd_file.close()

        if not success:
            return None, None
        return 'http://{0}:{1}/{2}.mpd'.format(
            _settings.httpd_listen(for_request=True),
            _settings.httpd_port(),
            self.video_id
        ), main_stream
```